### DISCUSSION TITLE
Could someone package for mac ?

### DISCUSSION THREAD

**Discussion by Doublejay** [qarmin/czkawka#1351] [Ideas]
I have no idea bout to get this working on mac.

**Comment by terenceongengkok** [comment]
hi razor, i'm a total noob at this.. and have tried to follow you instructions and failed miserably..

**when you create the .app it pops up with a notification - i assume keep the .app and the show content packages to enter folder.**

**when you get to textedit - i've made it plain text  but it keeps saving as a .txt file - how do i make it a .plist?**
-----------------------------------------------------------------------
Open Terminal: Open the Terminal application.

Navigate to the MacOS Folder: Use the cd command to navigate to the Contents/MacOS folder: - **do i just type the below in terminal?**

cd /path/to/Czkawka.app/Contents/MacOS
content_copy
download
Use code with caution.
Bash

(Please replace /path/to/Czkawka.app with the actual path to your Czkawka.app folder.) - **do you mean cd /path/to/MacintoshHD/Users/terenceo/Desktop/Czkawka.app/Contents/MacOS**

**Reply by RazorBackRoar** [reply]
> hi razor, i'm a total noob at this.. and have tried to follow you instructions and failed miserably..
> 
> 
> 
> **when you create the .app it pops up with a notification - i assume keep the .app and the show content packages to enter folder.**
> 
> 
> 
> **when you get to textedit - i've made it plain text  but it keeps saving as a .txt file - how do i make it a .plist?**
> 
> -----------------------------------------------------------------------
> 
> Open Terminal: Open the Terminal application.
> 
> 
> 
> Navigate to the MacOS Folder: Use the cd command to navigate to the Contents/MacOS folder: - **do i just type the below in terminal?**
> 
> 
> 
> cd /path/to/Czkawka.app/Contents/MacOS
> 
> content_copy
> 
> download
> 
> Use code with caution.
> 
> Bash
> 
> 
> 
> (Please replace /path/to/Czkawka.app with the actual path to your Czkawka.app folder.) - **do you mean cd /path/to/MacintoshHD/Users/terenceo/Desktop/Czkawka.app/Contents/MacOS**



Hey my friend I’d be happy to walk you through everything step by step to get CZKAWKA duplicates working. Do you have Discord? If so, we can go over it together live. If not, I’ll write out a clear, simple guide for you later tonight — just let me know what works best for you.

I’ve got the full process down — from the terminal commands to making the file executable. I use the nightly builds, but we can stick with the stable 9.0.0 release if you prefer. Once I walk you through it, you’ll see how simple it really is.

P.S. I actually found a much easier method — no need to turn it into an app. Making it an executable file is way faster and cleaner. From there, we can create a simple alias (like czk) in your terminal, and boom — just type it in and it launches instantly. No app wrapping, no extra steps.

We can give it the right permissions in macOS security settings, and the whole setup shouldn’t take more than 7–10 minutes. Personally, I stick with Zsh as my default shell (bash always drove me nuts), so all my setup instructions are written with Zsh in mind.

You’ll get all the features without the hassle. I’ll make sure it’s ready for you tonight — unless you’d rather jump on Discord in case you have any follow-up questions.

Let me know what works best for you!

**Reply by terenceongengkok** [reply]
hi razor, thanks so much for the help - i'll see how i go with your new process first!

between a newborn baby and time zone differences (i'm in australia), it might be a bit challenging to do a discord call at the moment haha

**Reply by RazorBackRoar** [reply]
> hi razor, thanks so much for the help - i'll see how i go with your new process first!
> 
> between a newborn baby and time zone differences (i'm in australia), it might be a bit challenging to do a discord call at the moment haha

Hey my friend ititss all taken care of for you. I hope you enjoy and if you have any questions please feel free to reach out.

**Reply by RazorBackRoar** [reply]
> hi razor, thanks so much for the help - i'll see how i go with your new process first!
> 
> between a newborn baby and time zone differences (i'm in australia), it might be a bit challenging to do a discord call at the moment haha

Perfect timing my friend. I just wrapped everything up and am now getting ready for the 4th of July celebrations. I hope you find the guide helpful, and please don't hesitate to reach out if you have any questions or need further assistance.

**Comment by RazorBackRoar** [comment]


**Reply by RazorBackRoar** [reply]
Hello, my friend — I hope this is straightforward for you. I’ve formatted everything in Markdown, so you can simply copy and paste as needed. If you have any questions or need anything further, feel free to reach out.

**Comment by RazorBackRoar** [comment]


**Reply by elistys** [reply]
I am following the instructions to install and use it on my M4 mac.
When I download mac_czkawka_gui_arm64 and run ./czkawka in the terminal, the following message appears and it won't start.

`dyld[8730]: Library not loaded: /opt/homebrew/opt/gtk4/lib/libgtk-4.1.dylib
  Referenced from: <B4D076C4-184A-3ED0-9231-AB6F7C09DB5A> /Users/***/Desktop/czkawka
  Reason: tried: '/opt/homebrew/opt/gtk4/lib/libgtk-4.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/opt/gtk4/lib/libgtk-4.1.dylib' (no such file), '/opt/homebrew/opt/gtk4/lib/libgtk-4.1.dylib' (no such file)
Abort trap: 6`

It seems like it's asking for other files to start up, how do I fix this?

**add note1**:I am getting the same error after running the below command.

```
# Install dependencies first
brew install gtk4 adwaita-icon-theme ffmpeg librsvg libheif libraw pkg-config

# Then install Czkawka
brew install --cask czkawka
```
**add note2**:I downloaded Nightly and the results were the same.

**Reply by RazorBackRoar** [reply]
# Hey! I ran into this at first and got it working by using the following method. 

## What's happening

You're getting this error when trying to run the app:
```
dyld[8730]: Library not loaded: /opt/homebrew/opt/gtk4/lib/libgtk-4.1.dylib
```

Basically, the app is looking for GTK4 (a graphics library) but can't find it on your Mac. Think of it like downloading a game that needs DirectX but you don't have it installed yet. The app needs certain files to run, and those files just aren't on your computer yet.

---

## The fix that worked for me

### Step 1: Make sure Homebrew is properly set up

First, check if Homebrew is installed:
```bash
brew --version
```

If it says "command not found", install Homebrew:
```bash
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

**Important:** After installing, Homebrew will show you two commands to run. Copy and paste those exactly. They look something like:
```bash
echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> ~/.zprofile
eval "$(/opt/homebrew/bin/brew shellenv)"
```

This tells your Mac where to find Homebrew. You only have to do this once.

---

### Step 2: Install GTK4 and dependencies

Once Homebrew is working, install the libraries:
```bash
brew install gtk4 adwaita-icon-theme
```

This takes a few minutes. What's happening here is you're downloading and installing the graphics files that the app needs to actually display its interface.

---

### Step 3: Make the file executable

Go to where you downloaded czkawka and make it executable:
```bash
cd ~/Desktop  # or wherever you put it
chmod +x czkawka_gui
```

This tells your Mac it's okay to run this file.

---

### Step 4: Try running it

```bash
./czkawka_gui
```

Should work now!

---

## Still not working? Try these fixes

### Fix 1: Install all possible dependencies
If you're still getting errors, install everything it might need:
```bash
brew install gtk4 adwaita-icon-theme gtksourceview5 librsvg pango gdk-pixbuf ffmpeg libheif libraw pkg-config
```

### Fix 2: Update gdk-pixbuf loaders
Sometimes the pixbuf loaders need to be regenerated:
```bash
gdk-pixbuf-query-loaders --update-cache
```

### Fix 3: Clear corrupted configs
If the app crashes on startup, try clearing the config files:
```bash
rm -rf "$HOME/Library/Application Support/pl.Qarmin.Czkawka"
rm -rf "$HOME/Library/Caches/pl.Qarmin.Czkawka"
```

Then try running again.

### Fix 4: Verify GTK4 installation
Check if GTK4 is actually installed correctly:
```bash
ls -la /opt/homebrew/opt/gtk4/lib/ | grep libgtk-4
```

You should see `libgtk-4.1.dylib` in the output. If not, try reinstalling:
```bash
brew uninstall gtk4
brew install gtk4
```

---

## Responding to your notes

**Note 1:** You said you're getting the same error after running:
```bash
brew install gtk4 adwaita-icon-theme ffmpeg librsvg libheif libraw pkg-config
brew install --cask czkawka
```

This is interesting because the cask version should handle dependencies automatically. Try this:
```bash
# Uninstall the cask version first
brew uninstall --cask czkawka

# Make sure GTK4 is installed
brew reinstall gtk4

# Reinstall the cask
brew install --cask czkawka
```

**Note 2:** You said the Nightly version had the same results.

Since both stable and nightly are failing, this points to a system-level issue with GTK4 or the library paths. Try running the app with debug output to see exactly what's failing:
```bash
RUST_BACKTRACE=full ./czkawka_gui
```

This will show more detailed error information that might help pinpoint the issue.

---

## Additional troubleshooting

### Check Homebrew's installation location
Apple Silicon Macs should have Homebrew at `/opt/homebrew/`. Verify this:
```bash
which brew
```

Should return `/opt/homebrew/bin/brew`. If it returns something else (like `/usr/local/bin/brew`), that's the Intel location and might be causing issues.

### Verify you downloaded the ARM64 version
Make sure the filename you downloaded says `arm64` or `aarch64` somewhere. The Intel (x86_64) version won't work properly on M4 Macs.

### Check file permissions
Sometimes macOS blocks downloaded executables. Try:
```bash
xattr -d com.apple.quarantine czkawka_gui
```

This removes the quarantine flag that macOS puts on downloaded files.

---

## Quick reference for common errors

**"brew: command not found"** → Run those two echo commands Homebrew showed after installation

**"Permission denied"** → Run `chmod +x czkawka_gui`

**"Library not loaded"** → Install GTK4 with `brew install gtk4`

**App crashes immediately** → Clear configs with the `rm -rf` commands above

**Still broken after everything** → Try the DMG version or build from source

---

Let me know if any of these fixes work for you or if you're still stuck. If nothing works, drop the output from `RUST_BACKTRACE=full ./czkawka_gui` and we can dig deeper into what's failing.

**Reply by elistys** [reply]
> Fix 5: Try the DMG version instead
Download the DMG installer from the [[releases page](https://github.com/qarmin/czkawka/releases)](https://github.com/qarmin/czkawka/releases) - those usually have dependencies bundled so you don't have to mess with any of this terminal stuff.

I can't find the DMG installer, can you post the direct link here?

**Reply by RazorBackRoar** [reply]
I Apologize I was looking at something else at the time ignore fix #5 everything is already bundled its just activating the scripts so they turn into executables. also I reworked the whole thing making it much nicer and accessible. this is the link https://gist.github.com/RazorBackRoar/72da40ac935961809722b45fcb5c86ae  its a gist now so I don't have to keep it as a repository. what issues are you still having?

**Reply by elistys** [reply]
Thanks, but I'm tired of reading the extensive documentation and struggling with Terminal, so I'll give up on installing it.

As an ordinary person, I don't understand why would distribute an app in a state where users have to assemble it using Terminal.

Is it extremely difficult for a developer (and even for you, a knowledgeable person) to distribute Czkawka as a finished, normal app, as the OP of this post is requesting?

**Reply by qarmin** [reply]
Because no one has fully completed the DMG file creation process, that’s why they simply don’t exist.

I personally use Linux (where this works almost out of the box or requires running just a single command to install external packages), and I’ve never had a Mac and don’t intend to have one - so I only provide the application itself, and users have to install it on their own (as you can see, different things work for different people, which makes creating unified documentation difficult).

There is a PR - https://github.com/qarmin/czkawka/pull/1252- but it is not integrated into CI and has not been tested by other people (who actually have a Mac), so it’s just waiting for someone to take it to completion.

I don’t know if I need to remind you, but there is a newer version of Czkawka - Krokiet - which does not require GTK dependencies, so it should be much easier to configure.

**Reply by elistys** [reply]
I hadn't considered that you don't have a Mac. That certainly makes it impossible to distribute the finished app.

That means RazorBackRoar is a developer who creates and distributes Mac apps, so it might be possible that he can complete the DMG file creation process completely.

Also, when I ran the same launch process as Czkawka with Krokiet, which you recommended,Krokiet launched without any errors! Duplicate file search also seems to work without any problems. Thank you.

However, the font size of the search results is very small, about 10pt, and very hard to read... It would be great if I could change the font size in the settings.

**Comment by soul-trad** [comment]
HI Razor,

Thanks for taking the time to write a guide for installing Czkawka as a single executable.  Sadly, although I can and have followed your instructions, when I open Czkawka from applications folder, I get a terminal window open up and. nothing else happens. The terminal window is below...    

I've tried installing from the terminal - no success - "no such cask" and tried various builds of the Mac silicon ( arm) Gui version as well as the package installer on the Czkawka website (doesn't seem to be a package file and none I can find on Github). Guessing that Mac is a b#tch to develop for and Apple does what it does best to ensure that anything not digitally signed for by Apple is going to be a PITA to run???  Any help would be appreciated! (not lest as DupGuru doesn't work reliably for me). 

Thanks again! 

(NB PC name and User name changed)

Last login: Wed Jul 23 17:22:24 on ttys000
/Applications/czkawka ; exit;
userrr@Userrrs-MacBook-Pro ~ % /Applications/czkawka ; exit;
17:23:00.027 [INFO] czkawka_core::common: Czkawka gtk version: 9.0.0, release mode, rust 1.85.0 (2025-02-17), os Mac OS 15.5.0 [aarch64 64-bit], 14 cpu/threads, features(1): [fast_image_resize], app cpu version: [Baseline], os cpu version: [Baseline]
17:23:00.028 [INFO] czkawka_core::common: Config folder set to "/Users/userrr/Library/Application Support/pl.Qarmin.Czkawka" and cache folder set to "/Users/userrr/Library/Caches/pl.Qarmin.Czkawka"

thread 'main' panicked at czkawka_gui/src/help_functions.rs:741:58:
Failed to create pixbuf from data: Error { domain: gdk-pixbuf-error-quark, code: 3, message: "Unrecognized image file format" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

thread 'main' panicked at library/core/src/panicking.rs:218:5:
panic in a function that cannot unwind
stack backtrace:
   0:        0x105ae754c - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h69fb0771ca4927e7
   1:        0x104e28a34 - core::fmt::write::ha746b8eed2a8b025
   2:        0x105accb94 - std::io::Write::write_fmt::h84907d310cf7838c
   3:        0x105ae7408 - std::sys::backtrace::BacktraceLock::print::h192522516a80a79a
   4:        0x105b04778 - std::panicking::default_hook::{{closure}}::h558a7c9eb7108e66
   5:        0x105b045f0 - std::panicking::default_hook::h0e1fd4244b979d73
   6:        0x105b04d7c - std::panicking::rust_panic_with_hook::h00b2262cd25a9e0f
   7:        0x105ae7a4c - std::panicking::begin_panic_handler::{{closure}}::h466628b32158069b
   8:        0x105ae7788 - std::sys::backtrace::__rust_end_short_backtrace::h328e872885259eca
   9:        0x105b04898 - _rust_begin_unwind
  10:        0x105c5e010 - core::panicking::panic_nounwind_fmt::h5be518e357a21bcc
  11:        0x105c5e088 - core::panicking::panic_nounwind::h1ce4397c6a41d91d
  12:        0x105c5e174 - core::panicking::panic_cannot_unwind::h4d9b851827ac7caa
  13:        0x1050e56d0 - gio::auto::application::ApplicationExt::connect_command_line::command_line_trampoline::h5abc20e6336d16b3
  14:        0x107186984 - __g_cclosure_marshal_INT__OBJECT
  15:        0x106b3a500 - _g_closure_invoke
  16:        0x106b4f0c8 - _signal_emit_unlocked_R
  17:        0x106b4e338 - _signal_emit_valist_unlocked
  18:        0x106b4dbb0 - _g_signal_emit_valist
  19:        0x106b4e6d4 - _g_signal_emit
  20:        0x1071d02d4 - _g_application_call_command_line
  21:        0x1071d1cb4 - _g_application_real_local_command_line
  22:        0x1071d0050 - _g_application_run
  23:        0x10515492c - gio::application::ApplicationExtManual::run_with_args::h9c6ff73e49eda111
  24:        0x105102fe8 - czkawka_gui::main::hdb4cdabbfddb5cd5
  25:        0x1050a5708 - std::sys::backtrace::__rust_begin_short_backtrace::h65be19c73093886e
  26:        0x10517da44 - std::rt::lang_start::{{closure}}::hf5f3671bd0ca3a6e
  27:        0x105ae62dc - std::rt::lang_start_internal::h518744cc10c95441
  28:        0x105106744 - _main
thread caused non-unwinding panic. aborting.
zsh: abort      /Applications/czkawka

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

**Reply by RazorBackRoar** [reply]
Hey Man,

The pixbuf error is likely due to missing or misconfigured gdk-pixbuf loaders or GTK 4 schemas. I'm updating the .md file now with the fix. Here's exactly what you need to do to resolve the pixbuf crash:

## 🎯 **Step-by-Step Fix:**

### 1. **Clear Corrupted Configs First:**
```bash
rm -rf "$HOME/Library/Application Support/pl.Qarmin.Czkawka"
rm -rf "$HOME/Library/Caches/pl.Qarmin.Czkawka"
```

### 2. **Try Running Again with Debug Info:**
```bash
cd /Applications
RUST_BACKTRACE=full ./czkawka
```

### 3. **If Still Crashes:**
The stable ARM build might be corrupted. Try the **nightly ARM build** instead:
- Go to: https://github.com/qarmin/czkawka/releases/tag/Nightly
- Download the ARM version
- Replace your current `/Applications/czkawka` file
- Run again

### 4. **If STILL Crashes:**
Use Homebrew as backup:
```bash
brew install gtk4 adwaita-icon-theme ffmpeg librsvg libheif libraw pkg-config
brew install --cask czkawka
```

## 🎯 **Most Likely Result:**
Step 1 (clearing configs) will probably fix it. The crash happens because Czkawka tries to load a corrupted config file on startup, and clearing those configs forces it to recreate them fresh.

If Step 1 doesn't work, then it's definitely a problem with that specific stable ARM build, and the nightly version should work better.

**Reply by RazorBackRoar** [reply]
I've updated the .md file above as well.

**Comment by RazorBackRoar** [comment]
> I have no idea bout to get this working on mac.

https://gist.github.com/RazorBackRoar/72da40ac935961809722b45fcb5c86ae  

This is the Gist I Created everything you need is right here.

**Comment by terenceongengkok** [comment]
I ended up just installing this cause I struggled too

https://www.cisdem.com/duplicate-finder.html
